Objects/TI4_PurgeBag.ttslua

```diff
@@ -3,14 +3,61 @@
 -- @author Jabberwocky
 --
 
+local _getByNameCache = {}
+function _getByName(name, tag)
+    local guid = _getByNameCache[name]
+    local object = guid and getObjectFromGUID(guid)
+    if object and ((not tag) or object.tag == tag) then
+        return object
+    end
+    for _, object in ipairs(getAllObjects()) do
+        if object.getName() == name and ((not tag) or object.tag == tag) then
+            _getByNameCache[name] = object.getGUID()
+            return object
+        end
+    end
+    
+    return false
+end
+
+function getRelicFragmentType(object)
+    return string.match(object.getName(), '^(%a+) Relic Fragment %(%d%)$')
+end
+
+function relocateRelicFragments(object)
+    local fragmentType = getRelicFragmentType(object)
+    if fragmentType then
+        local fragBag = _getByName(fragmentType .. ' Relic Fragments Bag', 'Bag')
+
+        if not fragBag then
+            return false
+        end
+
+        self.takeObject({
```

Comment:
should put this in a delayMove wrapper then Wait.frames() for at least 3 frames to move it.  onObjectEnterContainer is called BEFORE TTS has actually finished injecting it, so there's a race condition where doing a take right away can create a solid black object (I think more common on slower systems).

Reply:
Is directly calling `bag.putObject` from the callback_function safe? I understand the callback is supposed to be after it's finished spawning, but have you seen issues doing that as well?

Reply:
oh I missed it was in the callback, yep you're good.

Reply:
oh wait, hang on no.
it isn't safe to call "takeObject" because the onObjectEnterContainer hasn't finished putting the object into the container at that point.

Reply:
Putting together a candidate, will merge this and wrap the take.  Thanks!